modules/compactor/compactor.go

```diff
@@ -19,7 +20,17 @@ import (
 )
 
 const (
-	waitOnStartup = 90 * time.Second
+	// ringAutoForgetUnhealthyPeriods is how many consecutive timeout periods an unhealthy instance
+	// in the ring will be automatically removed.
+	ringAutoForgetUnhealthyPeriods = 2
+
+	// We use a safe default instead of exposing to config option to the user
+	// in order to simplify the config.
+	ringNumTokens = 512
+)
+
+var (
+	ringOp = ring.NewOp([]ring.InstanceState{ring.ACTIVE}, nil)
```

Comment:
nice +1 to this ringOp over the old one.

Reply:
Shamelessly taken from cortex :)